ledger/src/blockstore_processor.rs

```diff
@@ -103,7 +103,7 @@ fn execute_batch(
     transaction_status_sender: Option<TransactionStatusSender>,
     replay_vote_sender: Option<&ReplayVoteSender>,
 ) -> Result<()> {
-    let (tx_results, balances, inner_instructions) =
+    let (tx_results, balances, inner_instructions, transaction_logs) =
```

Comment:
Only API nodes really need to track transaction logs, validators don't need to. How about disabling log collection when `transaction_status_sender.is_none()`?